### PR TITLE
feat(EOF): Put EOF bytecode behind an Arc

### PR DESCRIPTION
**Pull request by jpgonzalezra**
This is the [ISSUE](https://github.com/bluealloy/revm/issues/1488)

DaniPopes: EOF structure is present inside of Bytecode, and is cloned internally in AccountInfo and other places. It would improve performance to put behind an Arc to avoid cloning the underlying memory.

cc: @rakita @DaniPopes

### CODE REVIEW

**Comment by jpgonzalezra** [comment]
I'm taking a look at why the tests are failing, it seems to be something else and not the changes made here

**Comment by rakita** [comment]
> I'm taking a look at why the tests are failing, it seems to be something else and not the changes made here

It is a malformed json file https://github.com/ethereum/tests/pull/1384

It is not related to this PR.

**Review by rakita — APPROVED** [review]
lgtm!

btw there are keywords that github understands that are good to use https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue

File: crates/interpreter/src/instructions/control.rs [hunk 1]
```diff
@@ -204,6 +204,8 @@ pub fn unknown<H: Host + ?Sized>(interpreter: &mut Interpreter, _host: &mut H) {
 
 #[cfg(test)]
 mod test {
+    use std::sync::Arc;
+
```
Comment (rakita) [inline crates/interpreter/src/instructions/control.rs:208]:
```suggestion
```

**Comment by jpgonzalezra** [comment]
Thanks for the recommendation, I will start using it.

**Pull request merged**